storage/client/types.go

```diff
@@ -2,10 +2,25 @@
 package client
 
 import (
+	"fmt"
 	"math/big"
 	"time"
 )
 
+type BigInt struct {
+	big.Int
+}
+
+func NewBigInt(v int64) BigInt {
+	var b BigInt
+	b.Set(big.NewInt(v))
+	return b
```

Comment:
I realized my original code was needlessly convoluted here; the entire function can be simply 
`return BigInt{*big.NewInt(v)}`

You can also use this syntax inside `validateBigInt()` and `numericToBigInt()`.